lib/DocumentManager.php

```diff
@@ -11,9 +11,26 @@
 
 namespace Sulu\Component\DocumentManager;
 
+use Sulu\Component\DocumentManager\Exception\DocumentManagerException;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
+/**
+ * A document manager has two roles:.
+ *
+ * - Provider of an API for working with documents.
+ * - A "sub container", providing access to the services which fall
+ *   within the scope of the particular document manager (as there can be many).
+ *
+ * This class is therefore an event dispatcher and a service container and has
+ * no other business logic within it.
+ *
+ * All primary methods are wrapped in a try catch block, the exception is then
+ * wrapped in a `DocumentManagerException` with a reference to this document
+ * managers name.
+ *
+ * TODO: Create a proxy document manager which automatically wraps all calls.
```

Comment:
Yes the `_call` method, which is the reason I did not implement it (also
the idea of having a `ExceptionWrappingDocumentManager` implementation).

On Thu, Apr 28, 2016 at 12:07:24AM -0700, Daniel Rotter wrote:
>    In [1]lib/DocumentManager.php:
> 
>  >  use Symfony\Component\EventDispatcher\EventDispatcherInterface;
>  >  use Symfony\Component\OptionsResolver\OptionsResolver;
>  >
>  > +/**
>  > + * A document manager has two roles:.
>  > + *
>  > + * - Provider of an API for working with documents.
>  > + * - A "sub container", providing access to the services which fall
>  > + *   within the scope of the particular document manager (as there can be many).
>  > + *
>  > + * This class is therefore an event dispatcher and a service container and has
>  > + * no other business logic within it.
>  > + *
>  > + * All primary methods are wrapped in a try catch block, the exception is then
>  > + * wrapped in a `DocumentManagerException` with a reference to this document
>  > + * managers name.
>  > + *
>  > + * TODO: Create a proxy document manager which automatically wraps all calls.
> 
>    So this proxy document manager would implement the _call method, or how
>    would you implement that?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r61381662
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61381662
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61381662